pandas/core/generic.py

```diff
@@ -5601,53 +5601,99 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
         """
         Trim values at input threshold(s).
 
+        Elements above the upper threshold will be changed to upper threshold.
+        Elements below the lower threshold will be changed to lower threshold.
+
         Parameters
         ----------
         lower : float or array_like, default None
+            Lower threshold for clipping.  Values smaller than upper will be
+            converted to lower.
         upper : float or array_like, default None
+            Upper threshold for clipping.  Values larger than upper will be
+            converted to upper.
         axis : int or string axis name, optional
             Align object with lower and upper along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        args : dictionary of arguments arguments passed to pandas.compat.numpy
+        kwargs : dictionary of keyword arguments passed to pandas.compat.numpy
 
         Returns
         -------
-        clipped : Series
+        clipped : DataFrame/Series
+            Elements above or below the upper and lower thresholds converted to
+            threshold values.
+
+        Notes
+        -----
+        Clipping data is a method for dealing with dubious elements.
+        If some elements are too large or too small, clipping is one way to
+        transform the data into a reasonable range.
+
+        See Also
+        --------
+        pandas.DataFrame.clip_upper : Return copy of input with values
+            above given value(s) truncated.
```

Comment:
add Series.clip

Reply:
this is in generic, isn't it being reused by Series.clip?